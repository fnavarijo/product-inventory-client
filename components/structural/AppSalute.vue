<template>
  <div class="text-dark-gray-700 text-right border-r border-dark-gray-600 pr-4">
    <span class="font-bold block">Bienvenido</span>
    {{ userName }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapState } from 'vuex';

export default Vue.extend({
  name: 'AppSalute',
  computed: {
    ...mapState('user', {
      userName: 'name',
    }),
  },
});
</script>
