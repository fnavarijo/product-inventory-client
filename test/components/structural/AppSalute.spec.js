import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import AppSalute from '@/components/structural/AppSalute.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('AppSalute component', () => {
  let store;

  beforeAll(() => {
    store = new Vuex.Store({
      modules: {
        user: {
          state: {
            name: 'John',
          },
          namespaced: true,
        },
      },
    });
  });

  it('should render salute for the user', () => {
    const wrapper = mount(AppSalute, {
      localVue,
      store,
    });

    expect(wrapper.html()).toContain('John');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
