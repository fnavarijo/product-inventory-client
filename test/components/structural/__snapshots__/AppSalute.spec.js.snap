// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSalute component should render salute for the user 1`] = `
<div class="text-dark-gray-700 text-right border-r border-dark-gray-600 pr-4"><span class="font-bold block">Bienvenido</span>
  John
</div>
`;
